### PR TITLE
Makefile: remove `format` command in favor of simpler `format-check`

### PR DESCRIPTION
**Pull request by lschuermann**
### Pull Request Overview

This removes the `make format` command which, in essence, runs some bash magic to achieve the exact same job that `cargo fmt` does, if its run within workspace instead of a particular crate. Besides this, many people's editors have a native rustfmt integration anyways, which makes this script and its prompt for unstaged files largely redundant.

Instead, we replace it with the more useful `make format-check` and the associated `tools/check_format.sh`. This script does more than `cargo fmt --check` does, for instance also ensuring that no Rust source file contains tab characters (which rustfmt sometimes leaves behind).

As part of `make prepush`, `check-format` is now more in line with the other jobs, such as `cargo check` or Clippy: it'll complain, but not touch any source code.

This also removes an optimization that uses Make's logic to check for changed files in order to skip formatting. However, I doubt that this will meaningfully impact users (`make format-check` completes in about 2sec and does not take a meaningful amount of time compared to the other jobs executed as part of `prepush`).

### Testing Strategy

CI / running make commands.

### TODO or Help Wanted

N/A

### Documentation Updated

- [ ] Updated the relevant files in `/docs`, or no updates are required.

### Formatting

- [x] Ran `make prepush`.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -174,15 +174,11 @@ clean:
...
-fmt format: tools/.format_fresh
-	$(call banner,Formatting complete)
-
-# Get a list of all rust source files (everything fmt operates on)
-$(eval RUST_FILES_IN_TREE := $(shell (git ls-files | grep '\.rs$$') || find . -type f -name '*.rs'))
-tools/.format_fresh: $(RUST_FILES_IN_TREE)
-	@./tools/run_cargo_fmt.sh $(TOCK_FORMAT_MODE)
-	@touch tools/.format_fresh
+.PHONY: fmt format-check
+fmt format-check:
```
Comment (ppannuto) [inline Makefile:178]:
```suggestion
format-check:
```

or, if we're committed to keeping a shorthand `fmt-chk`, but `fmt` alone is a verb that formats IMHO, which this no longer does IIUC

**Review by ppannuto — COMMENTED** [review]
This seems fine to me.

I have one concern in current form. If I understand correctly, we can be in a situation where `cargo fmt` runs but then `make format-check` fails, because of tab characters? It seems a bit of a regression that there's no command I can run that results in "force everything to conform to the format checker".

Might still need a wrapper bash script around `cargo fmt`, albeit a simpler one, for the tab issue?

File: Makefile (hunk 1 above)
Comment (lschuermann) [reply Makefile:178]: Right, I completely missed this shorthand. Happy to update.

**Comment by lschuermann** [comment]
> Might still need a wrapper bash script around `cargo fmt`, albeit a simpler one, for the tab issue?

It's a rare enough issue that I don't think we'll hit it much in practice, but I'll let the jury decide.

**Review by bradjc — CHANGES_REQUESTED** [review]
I need `make fmt` to be an alias for `cargo fmt`. We might as well also keep `make format` as well.

**Comment by bradjc** [comment]
> > Might still need a wrapper bash script around `cargo fmt`, albeit a simpler one, for the tab issue?
> 
> It's a rare enough issue that I don't think we'll hit it much in practice, but I'll let the jury decide.

I think as long as CI ensures it's fixed we're fine.

**Comment by bradjc** [comment]
> I need `make fmt` to be an alias for `cargo fmt`. We might as well also keep `make format` as well.

Those commands date to 2016: d4f62cfd20ca8f11a376d1e2556b671791e84683 (Also wow my attitude on commands has changed! Now I say shorter the better! But code should be descriptive.)

**Comment by lschuermann** [comment]
> I need `make fmt` to be an alias for `cargo fmt`. We might as well also keep `make format` as well.

Sure, will re-add. From the discussion in the slack `#core` channel I reckon that for this explicit command (when it's not invoked as part of another) we don't need the git prompt though?

In that case I'll literally alias it to `cargo fmt`, which will do the right thing(TM).

In the future this `make fmt` command will also allow us to add additional formatting steps, like removing tab characters (maybe, if we're confident this can't break anything), reflowing Markdown, adding TOCs, etc.

**Comment by lschuermann** [comment]
Should be good to go then!

File: tools/check_format.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
+#!/usr/bin/env bash
+
+# Licensed under the Apache License, Version 2.0 or the MIT License.
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+# Copyright Tock Contributors 2023.
+#
+# Author: Leon Schuermann <leon@is.currently.online>
+# Author: Pat Pannuto <pat.pannuto@gmail.com>
+# Author: Brad Campbell <bradjc5@gmail.com>
```
Comment (bradjc) [inline tools/check_format.sh:9]:
```suggestion
# Licensed under the Apache License, Version 2.0 or the MIT License.
# SPDX-License-Identifier: Apache-2.0 OR MIT
# Copyright Tock Contributors 2023.
#
# Runs `rustfmt --check` on the workspace and check for tabs in Rust files.
# Must be run from root Tock directory.
#
# Author: Leon Schuermann <leon@is.currently.online>
# Author: Pat Pannuto <pat.pannuto@gmail.com>
# Author: Brad Campbell <bradjc5@gmail.com>
```

**Review by bradjc — APPROVED** [review]

**Review by ppannuto — APPROVED** [review]

**Pull request merged**